lib/OpenQA/Schema/Result/JobGroups.pm

```diff
@@ -166,12 +166,25 @@ sub _find_expired_jobs {
 
     # all jobs not in important builds that are expired
     my $timecond = {'<' => time2str('%Y-%m-%d %H:%M:%S', time - 24 * 3600 * $keep_in_days, 'UTC')};
-    push(@ors, {BUILD => {-not_in => $important_builds}, t_finished => $timecond});
+
+    # filter out linked jobs
+    my $expired_jobs = $self->jobs->search({BUILD => {-not_in => $important_builds}, t_finished => $timecond}, {order_by => qw(id)});
+    my @linked_jobs = ();
+    while (my $job = $expired_jobs->next) {
+        my $comments = $job->comments;
+        while (my $comment = $comments->next) {
+            if ($comment->label =~ /^linked/) {
```

Comment:
we actually want all labeled tests to be considered important. Linked is just the automatic part

And your implementation is horrible because you go through tons of jobs and query the comments table one by one. You need some kind of join here, i.e. to query only comments if job_id => in (SUBQUERY)